src/ram_key_value_store/mod.rs

```diff
@@ -75,6 +81,17 @@ impl<'a> KeyValueStore for RamKeyValueStore<'a> {
             if key == element.key {
                 assert!(mem::size_of::<T>() == element.size as usize, "Invalid size");
 
+                let value_slice = unsafe {
+                    core::slice::from_raw_parts(
+                        value as *const T as *const Cell<u8>,
+                        element.size as usize,
+                    )
+                };
+
+                if value_slice != &self.ram[offset..] {
```

Comment:
shouldn't we match the slice sizes? because this should be comparing everything in `ram` after the offset... so really it should be `&self.ram[offset..offset+element.size]` no?

Reply:
Oops, you're right